src/extension.ts

```diff
@@ -73,6 +73,39 @@ export const readReplicas = (options: ReplicasOptions, configureReplicaClient?:
             if (model) {
               return replica[model][operation](args)
             }
+
+            if (operation === '$queryRaw') {
+              const rows = await replica[operation](args)
+              // HACK: Push into the mapped format Prisma expects when rehydrating
+              // result types.
+              // https://github.com/prisma/prisma/issues/21139
+              rows.forEach((row: Record<string, unknown>) => {
+                Object.keys(row).forEach((key) => {
+                  row[key] = {
+                    prisma__type: undefined,
+                    prisma__value: row[key],
+                  }
+                })
+              })
```

Comment:
Do you agree that this change belongs in the ORM? If so, would you be willing to contribute a PR for us there?

Reply:
Would it potentially be acceptable to add this here until the fix is in ORM though?

(Of course if this can be merged at all depends a lot on the other comment I left)

Reply:
I'd be willing to try in the future, but I don't suspect I'll have capacity to open up and onboard onto the full client codebase in the next month or so. This solution works for now without downstream effects, so is a lot less risky to use in the short term.